### PR TITLE
Allow to remove all archived entries

### PR DESCRIPTION
**Pull request by tcitworld**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documentation | no
| Translation   | yes
| Fixed tickets | #2948
| License       | MIT

### CODE REVIEW

**Comment by tcitworld** [comment]
Was targetting master at first, may need rebase.

**Comment by tcitworld** [comment]
Hmm, Travis issue is strange.

**Comment by nicosomb** [comment]
I launched testsuite locally on sqlite and mysql, and no issues. @j0k3r any idea?

**Comment by j0k3r** [comment]
This is strange :thinking:

File: tests/Wallabag/CoreBundle/Controller/ConfigControllerTest.php [hunk 1]
```diff
@@ -862,7 +862,7 @@ public function testResetArchivedEntries()
 
         $entryReset = $em
             ->getRepository('WallabagCoreBundle:Entry')
-            ->countAllEntriesByUsername($user->getId());
```
Comment (tcitworld) [inline tests/Wallabag/CoreBundle/Controller/ConfigControllerTest.php]: lol

**Comment by tcitworld** [comment]
👍

File: src/Wallabag/AnnotationBundle/Repository/AnnotationRepository.php [hunk 2]
```diff
@@ -122,4 +122,21 @@ public function removeAllByUserId($userId)
...
     }
+
+    /**
+     * Find all annotations related to archived entries.
+     *
+     * @param $userId
+     *
+     * @return mixed
+     */
+    public function findAllByArchivedEntriesAndUserId($userId)
```
Comment (j0k3r) [inline src/Wallabag/AnnotationBundle/Repository/AnnotationRepository.php]: `findAllArchivedEntriesByUser` seems more appropriate

File: src/Wallabag/CoreBundle/Repository/TagRepository.php [hunk 3]
```diff
@@ -76,4 +76,24 @@ public function findOneByEntryAndTagLabel($entry, $label)
             ->getQuery()
             ->getSingleResult();
     }
+
+    public function findTagsForArchivedArticles($userId)
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Repository/TagRepository.php]: `findForArchivedArticlesByUser` ?

**Comment by arooni** [comment]
where is this option in the UI? 

also, does anyone know how many frree articles framabag supports?

**Pull request merged**